src/os_crypto/shared/keys.c

```diff
@@ -515,8 +523,12 @@ void OS_PassEmptyKeyfile() {
 
 /* Delete a key */
 int OS_DeleteKey(keystore *keys, const char *id, int purge) {
-    int i = OS_IsAllowedID(keys, id);
+    if (keys->flags.key_mode != W_RAW_KEY && keys->flags.key_mode != W_DUAL_KEY) {
```

Comment:
Shouldn't this be an `OR` operation? I reviewed the assignment of the `key_mode` and it always has just one value (`W_RAW_KEY`, `W_ENCRYPTION_KEY` or `W_DUAL_KEY`).
If not, in any case, the condition would be more clear if you just evaluate `keys->flags.key_mode != W_ENCRYPTION_KEY`... That's correct?

Reply:
The **OR** operation is correct because it's checking if the key store **isn´t** in RAW mode **and** **isn´t** in DUAL mode.
It's true that it can be resolved with a unique comparison with W_ENCRYPTION_KEY.
But, this logic was considered as more sustainable. If, in the future, we add a fourth initialization mode, comparing with keys->flags.key_mode != W_ENCRYPTION_KEY will be erroneous.
In this way, we know that this method can only be used with the two modes that are compared here.

Reply:
Good! That makes sense!

Thanks for the clarification on the criteria that you followed for this implementation.